zk.el

```diff
@@ -191,10 +191,10 @@ Used in conjunction with `format', the string `%s' will be
 replaced by a note's ID."
   :type 'string)
 
-;; This needs to be a macro in order to reflect user changes to the variables.
-(defmacro zk-link-regexp ()
-  "Returns regexp for zk-link based on `zk-link-format' and `zk-id-regexp'."
-  '(format (regexp-quote zk-link-format) zk-id-regexp))
+(defun zk-link-regexp ()
```

Comment:
It's not clear to me what difference it makes if this is a function vs a macro. Do you know the implications here?

Reply:
The implication is that a macro generates code that is then simply inserted wherever the macro is called from, so there is no run-time performance difference at all between using a macro and just copy/pasting the code it generates. However, if the defmacro definition is re-evaluated, the code that used that macro will not be updated until it is compiled or evaluated again:

``` emacs-lisp
(defmacro foo (bar)
  "Foo that bar."
  `(list 'fooing 'that ,bar))
(foo 'lamp) ;; => (fooing that lamp)
(defun foobar (baz)
  "Foo the bar with baz."
  (foo baz))
(foobar 'cup) ;; => (fooing that cup)
(defmacro foo (bar)
  "Foo that bar differently."
  `(list 'fooing 'that ,bar 'differently))
(foo 'lamp) ;; => (fooing that lamp differently)
(foobar 'book) ;; => (fooing that book)
```

So, not really a big deal for end use, but can make coding and debugging a pain if you forget to re-evaluate the functions that use the macro that you changed.